packages/api/src/utils/pin.js

```diff
@@ -135,12 +135,18 @@ export async function waitOkPins (cid, cluster, waitTime = MAX_PIN_STATUS_CHECK_
  * @param {import('@web3-storage/db').DBClient} db
  * @param {number} waitTime
  * @param {number} checkInterval
- * @return {Promise.<import('@web3-storage/db/db-client-types').PinUpsertInput[]>}
+ * @return {Promise.<import('@web3-storage/db/db-client-types').PinsUpsertInput[]>}
  */
 export async function waitAndUpdateOkPins (cid, cluster, db, waitTime = MAX_PIN_STATUS_CHECK_TIME, checkInterval = PIN_STATUS_CHECK_INTERVAL) {
   const okPins = await waitOkPins(cid, cluster, waitTime, checkInterval)
-  for (const pin of okPins) {
-    await db.upsertPin(cid, pin)
-  }
-  return okPins
+  const pins = okPins.map((pin) => {
+    return {
+      _id: pin._id,
+      status: pin.status,
+      cid,
+      locationId: pin.location._id
+    }
+  })
+  await db.upsertPins(pins)
+  return pins
```

Comment:
Any reason for changing the return value?

Reply:
Not specifically, so I have reverted it.